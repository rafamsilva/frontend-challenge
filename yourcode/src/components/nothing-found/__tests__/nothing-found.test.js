import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import NothingFound from '../';

describe('Nothing found Component', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  describe('Render', () => {
    it('should render without crashing', () => {
      act(() => {
        render(<NothingFound />, container);
      });
    });

    it('should render with default values', () => {
      act(() => {
        render(<NothingFound />, container);
      });

      expect(container.querySelector("[data-testid='title']")).toBeDefined();
      expect(container.querySelector("[data-testid='title']").textContent).toBe('Don’t know what to search?');

      expect(container.querySelector("[data-testid='description']")).toBeDefined();
      expect(container.querySelector("[data-testid='description']").textContent).toBe('Here’s an offer you can’t refuse');
    });

    it('should render with passed values', () => {
      act(() => {
        render(<NothingFound terms="Darwin" />, container);
      });

      expect(container.querySelector("[data-testid='title']")).toBeDefined();
      expect(container.querySelector("[data-testid='title']").textContent).toBe('Sorry, nothing found!');

      expect(container.querySelector("[data-testid='description']")).toBeDefined();
      expect(container.querySelector("[data-testid='description']").textContent).toBe(
        `Sorry, nothing was found to term: \`Darwin\`. Try again with different terms`,
      );
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot default values', () => {
      act(() => {
        render(<NothingFound />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });

    it('should match snapshot passed values', () => {
      act(() => {
        render(<NothingFound terms="Darwin" />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
  });
});
