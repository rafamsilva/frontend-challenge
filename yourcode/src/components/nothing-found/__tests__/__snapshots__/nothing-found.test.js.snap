// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nothing found Component Snapshot should match snapshot default values 1`] = `
"<div class=\\"sc-AxjAm lcVokx\\"><svg alt=\\"Illustration\\" title=\\"Illustration\\" class=\\"sc-AxirZ fHEVYp\\">empty-illustration.svg</svg>
  <h3 data-testid=\\"title\\" class=\\"sc-AxiKw QDhxu\\">Don’t know what to search?</h3>
  <p data-testid=\\"description\\" class=\\"sc-AxhCb iiHvbn\\">Here’s an offer you can’t refuse</p>
</div>"
`;

exports[`Nothing found Component Snapshot should match snapshot passed values 1`] = `
"<div class=\\"sc-AxjAm lcVokx\\"><svg alt=\\"Illustration\\" title=\\"Illustration\\" class=\\"sc-AxirZ fHEVYp\\">empty-illustration.svg</svg>
  <h3 data-testid=\\"title\\" class=\\"sc-AxiKw QDhxu\\">Sorry, nothing found!</h3>
  <p data-testid=\\"description\\" class=\\"sc-AxhCb iiHvbn\\">Sorry, nothing was found to term: \`Darwin\`. Try again with different terms</p>
</div>"
`;
